src/main/java/dataMgmt/MetadataManager.java

```diff
@@ -156,6 +184,29 @@ public void fileMoved(long fileID, String newFilePath) {
         fileIDtoFilePath.put(fileID, newFilePath);
     }
 
+    public void fileDeleted(long fileID) {
+        String filePath = fileIDtoFilePath.get(fileID);
+        if (filePath != null){
+            fileMetadataMap.remove(filePath);
+        }
+        fileIDtoFilePath.remove(fileID);
+
+        long id = fileIDtoProjectID.get(fileID);
```

Comment:
check out http://stackoverflow.com/questions/6588856/conversion-from-null-to-int-possible